zeth-contracts/contracts/BaseMixer.sol

```diff
@@ -189,6 +191,10 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
             digest_inputs[0] = primary_inputs[i]; // See the way the inputs are ordered in the extended proof
             digest_inputs[1] = primary_inputs[i+1];
             bytes32 current_commitment = Bytes.sha256_digest_from_field_elements(digest_inputs);
+            require(
+                uint256(current_commitment) < r,
```

Comment:
Ah sorry I removed this check from the assemble function and forgot to add it again (c.f. last commit) (since the assemble function should disappear with the new packing policy).

It is written in the issue to check the primary inputS so I thought you also might want to check tall of them.

Reply:
Please make sure to double check the diffs of your PRs next time

Reply:
You were not checking the right thing here.
In the lines commented above you compare a 256-bit number with the scalar field characteristic `r` (encoded on 253 bits), I think you haven't ran the python tests as I'd bet they would fail most of the time (nothing should prevent your nullifiers and commitments, both output of sha256 or blake2s to be smaller than `r`, why did you think they should?)

Reply:
Yeah I saw some tests failing.
I thought that these commitments were the field commitments already and not the blake2s digests.

Reply:
So do I understand correctly that you wanted to merge to develop a piece of code that:
1. Does not fix the vulnerability
2. Breaks the tests
??
Please make sure to review the other PRs you have open.